protocol/index.ts

```diff
@@ -50,6 +50,24 @@ export const ArcjetEmailType: ArcjetEnum<ArcjetEmailType> = {
   INVALID: "INVALID",
 };
 
+export type SensitiveInfoEntity<Custom extends string> =
+  | "email"
+  | "phone-number"
+  | "ip-address"
+  | "credit-card-number"
+  | Custom;
+
+export type IdentifiedEntity<Custom extends string> = {
+  start: number;
+  end: number;
+  identifiedType: SensitiveInfoEntity<Custom>;
+};
+
+export type DetectSensitiveInfoResult<Custom extends string> = {
+  allowed: IdentifiedEntity<Custom>[];
+  denied: IdentifiedEntity<Custom>[];
+};
```

Comment:
These aren't user-facing types so they shouldn't exist in the protocol. There's also no real reason to be creating new types for analyze when the generated types work pretty well.